### PR TITLE
DataOps for LLMOps - AML Sample

### PR DESCRIPTION
**Pull request by ra1han**
Purpose of this PR is to add the first DataOps sample using AML

The PR introduces the following new artefacts -

DataOps folder: includes the scripts used by the CI/CD pipelines.
named_entity_recognition/data_pipelines folder: a sample data pipeline to prepare the dataset for the NER use case
named_entity_recognition_data_aml_cd_workflow.yml: this is the github CD pipeline under the .github/workflows folder

### CODE REVIEW

File: docs/how_to_configure_dataops.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+## The Sample Implementation
+
+This repository includes an implementation of DataOps for the `named_entity_recognition` sample. The sample implementation uses Azure Machine Learning to run the data pipelines.
+
+The data pipeline loads data from the source system, processes it, and stores it in the target location. The processed data is stored as JSONL files, which are registered as data assets
+
+![dataops llmops](images/dataops_llmops.png)
+
+The sample data pipelines assume that raw data is stored in a storage account. The raw data is processed and transformed into clean JSONL files and stored in the target storage account.
```
Comment (ritesh-modi) [inline docs/how_to_configure_dataops.md:19]: This line is redundant

File: docs/how_to_configure_dataops.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+If you are not using data pipelines to create the data assets, the Promptflow flows will use the JSONL files inside the `data` folder to create the data assets.
+
+## Steps to Configure DataOps
+
+Follow these steps to configure DataOps for your flow:
+
+**New Folder for data pipelines** The data pipelines for the `named_entity_recognition` flow are inside a sub-folder named `data_pipelines`. Create a similar folder under your flow folder for the data pipelines.
+
+**Configure source and target location** As mentioned earlier, the data pipeline loads data from a source storage account container and stores the processed data in a target storage account container. Create these two containers and upload the source dataset to the source container.
```
Comment (ritesh-modi) [inline docs/how_to_configure_dataops.md:31]: the target should be AML data assets?

File: docs/how_to_configure_dataops.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+**Data Pipeline Configuration:** The `dataops_config.json` file contains configurations for the data pipeline.
+
+You can start by copying an existing config file and modify it with relevant values. Provide valid values for all the configuration elements.
+
+**Updating Flow Configuration:** The configuration of the use-case is managed by the `experiment.yaml` (sets the flow paths, datasets, and evaluations). The `experiment.yaml` in the repo uses local data files. If you are using DataOps,this config file needs to point to the Data Asset path. The data asset path will look like this `azureml://datastores/[data_store_name]/paths/[file_path]`
+
+So, update any `datasets` component in the `experiment.yaml` file and make sure the `source` field point to the Data Asset path.
```
Comment (ritesh-modi) [inline docs/how_to_configure_dataops.md:39]: "So, update any `datasets` component in the `experiment.yaml` file and make sure the `source` field point to the Data Asset path." line can be re-witten as "Update any `datasets` elements in the `experiment.yaml` files and make sure the `source` field point to the Data Asset path in format azureml:<my_data_asset_name>:<version>.

**Review by ritesh-modi — CHANGES_REQUESTED** [review]

File: docs/how_to_configure_dataops.md (hunk 1 above)
Comment (ra1han) [reply docs/how_to_configure_dataops.md:19]: Right.

File: docs/how_to_configure_dataops.md (hunk 2 above)
Comment (ra1han) [reply docs/how_to_configure_dataops.md:31]: The physical file is stored in the storage account. The storage account folder is later mapped as AML data asset. I have added another line to make it obvious.

File: docs/how_to_configure_dataops.md (hunk 3 above)
Comment (ra1han) [reply docs/how_to_configure_dataops.md]: Updated.

**Review by ritesh-modi — APPROVED** [review]
LGTM. We have reviewed it offline as well

**Pull request merged**